app/src/main/java/camp/nextstep/edu/github/RepositoryAdapter.kt

```diff
@@ -0,0 +1,49 @@
+package camp.nextstep.edu.github
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import camp.nextstep.edu.github.databinding.ItemRepoBinding
+import camp.nextstep.edu.github.domain.model.GitHubRepositoryData
+
+class RepositoryAdapter : ListAdapter<GitHubRepositoryData, RepositoryViewHolder>(RepositoryComparator.COMPARATOR) {
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RepositoryViewHolder {
+        return RepositoryViewHolder.create(parent)
+    }
+
+    override fun onBindViewHolder(holder: RepositoryViewHolder, position: Int) {
+        val repository = getItem(position)
+        holder.bind(repository)
+    }
+}
+
+class RepositoryViewHolder(private val binding: ItemRepoBinding) : RecyclerView.ViewHolder(binding.root) {
+    companion object {
+        fun create(parent: ViewGroup): RepositoryViewHolder {
+            val layoutInflater = LayoutInflater.from(parent.context)
+            val binding = ItemRepoBinding.inflate(layoutInflater, parent, false)
+            return RepositoryViewHolder(binding)
+        }
+    }
+
+    fun bind(repositoryData: GitHubRepositoryData) {
+        binding.itemRepoTvFullname.text = repositoryData.fullName
+        binding.itemRepoTvDescription.text = repositoryData.description
+    }
+}
+
+object RepositoryComparator {
+    val COMPARATOR by lazy {
+        object : DiffUtil.ItemCallback<GitHubRepositoryData>() {
+            override fun areItemsTheSame(oldItem: GitHubRepositoryData, newItem: GitHubRepositoryData): Boolean {
+                return oldItem.fullName == oldItem.fullName
+            }
+
```

Comment:
싱글턴 안에 익명 클래스로 두는 것 보다는, 아예 DiffUtil 자체를 싱글턴 오브젝트로 만들어 보는 것은 어떨까요?
```suggestion
object RepositoryComparator : DiffUtil.ItemCallback<GitHubRepositoryData>() {
        override fun areItemsTheSame(oldItem: GitHubRepositoryData, newItem: GitHubRepositoryData): Boolean {
            return oldItem.fullName == oldItem.fullName
        }
.....
```